### PR TITLE
Added onchain actions: tx, vote, stake.

### PR DESCRIPTION
**Pull request by romfrolov**

### CODE REVIEW

File: core/Account.js [hunk 1]
```diff
@@ -33,62 +40,87 @@ function Account() {
...
+ * @param  {Number} amount  Amount to coins to use for vote.
+ * @return {String}         Serialized signed transaction.
  */
-Account.prototype.vote = function vote(delegateAddress) {
-    const hash = keccak256(delegateAddress);
-    const sig  = helpers.generateSignature(hash, this.secretKey);
-    const msg  = {delegateAddress, sig};
+Account.prototype.vote = function vote(address, amount) {
+    const data = helpers.encodeTxData('vote(address,uint256)', [address, amount]);
+    const tx   = new ethereumTx({nonce: this.nonce, to: ZERO_ADDRESS, value: amount, data});
```
Comment (damirka) [inline core/Account.js]:
```suggestion
    return this.tx(ZERO_ADDRESS, amount, data);
```

File: lib/helpers.js [hunk 2]
```diff
@@ -25,24 +24,35 @@ exports.generateSignature = function generateSignature(dataToSign, secretKey) {
...
+ * @return {String}          Hexadecimal tx data string with method signature and encoded values.
  */
-exports.sign = function sign(secretKey, tx, chainId) {
-    const encodedData = rlp.encode(JSON.stringify(tx));
-    const msgHash = keccak256(encodedData);
-    const sig = secp256k1.sign(msgHash, secretKey);
-    const recovery = sig.recovery;
+exports.encodeTxData = function encodeTxData(method, values) {
+    return values.length === 0
+        && '0x' + keccak256(method).toString('hex').slice(0, 8)
```
Comment (damirka) [inline lib/helpers.js]: Why 8? Should be a constant or description. Otherwise it's a magic number.

File: core/Account.js [hunk 3]
```diff
@@ -33,62 +40,87 @@ function Account() {
...
-    // TODO
-    // 1. Mark amount as locked (locking mechanism?).
-    // 2. Generate certificates (random?).
-    // 3. transport.send(stakeRequest?);
+Account.prototype.stake = function stake(amount) {
+    const data = helpers.encodeTxData('stake(uint256)', [amount]);
+    const tx   = new ethereumTx({nonce: this.nonce, to: ZERO_ADDRESS, value: amount, data});
+    tx.sign(this.secretKey);
+
+    return '0x' + tx.serialize().toString('hex');
```
Comment (damirka) [inline core/Account.js]: Same as above. Use Account~tx instead.

**Review by damirka — CHANGES_REQUESTED** [review]

File: lib/helpers.js (hunk 2 above)
Comment (romfrolov) [reply lib/helpers.js]: 4 bytes like default Ethereum method signature.

**Review by damirka — APPROVED** [review]

**Pull request merged**